### PR TITLE
autogen 5.18.16

### PR DESCRIPTION
**Pull request by alebcay**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by mistydemeo** [comment]
Patch doesn't apply to this release. Can you check if it's still necessary?

**Comment by alebcay** [comment]
The `configure` script now looks for `guile` 2.2 as well as the previous 2.0 and 1.8, so it seems that the patch is no longer necessary.

**Comment by mistydemeo** [comment]
Thank you!

Looks like there's an issue with `mktemp` now:

```
==> make
/bin/sh config/mk-shdefs config/shdefs
+ exec
++ mktemp --suffix=.tdir -d /tmp/shdefs-XXXXXXXXX
mktemp: illegal option -- -
usage: mktemp [-d] [-q] [-t prefix] [-u] template ...
       mktemp [-d] [-q] [-u] -t prefix 
+ tmp=
+ exec
config/mk-shdefs: line 134: /mk-shdef.log: Operation not permitted
make: *** [config/shdefs] Error 1
```

I think they're using GNUisms; BSD `mktemp` doesn't have double dash options and doesn't have a suffix option.

**Comment by mistydemeo** [comment]
Upstream bug report: https://sourceforge.net/p/autogen/bugs/189/ It's not resolved yet. We may have to patch this.

**Comment by mistydemeo** [comment]
We can fix this for the time being by adding `depends_on "coreutils => :build`, then in the install method, `inreplace %w[agen5/mk-stamps.sh build-aux/run-ag.sh config/mk-shdefs.in], "mktemp", "gmktemp"`.

**Comment by alebcay** [comment]
Gotcha. Would the preferred solution be to depend on `coreutils` for build or to patch out `mktemp --suffix=.tdir ...` invocations with `mktemp ...`?

**Comment by mistydemeo** [comment]
Let's use coreutils; I feel slightly more nervous about the latter, because then we're doing stuff the buildsystem wasn't tested against. Thanks!

File: Formula/autogen.rb [hunk 1]
```diff
@@ -13,16 +12,12 @@ class Autogen < Formula
...
   depends_on "guile"
 
-  # Allow guile 2.2 to be used
-  patch do
-    url "https://raw.githubusercontent.com/Homebrew/formula-patches/0de886b/autogen/allow-guile-2.2.diff"
-    sha256 "438fe673432c96d5da449b84daa4d1c6ad238ea0b4ccd13491872df8c51fa978"
-  end
-
   def install
+    inreplace %w[agen5/mk-stamps.sh build-aux/run-ag.sh config/mk-shdefs.in], "mktemp", "gmktemp"
```
Comment (mistydemeo) [inline Formula/autogen.rb]: Thanks a lot for this! Can you include a link to the bug report above it? e.g.

```suggestion
    # Uses GNU-specific mktemp syntax: https://sourceforge.net/p/autogen/bugs/189/
    inreplace %w[agen5/mk-stamps.sh build-aux/run-ag.sh config/mk-shdefs.in], "mktemp", "gmktemp"
```

**Review by mistydemeo — CHANGES_REQUESTED** [review]

**Review by mistydemeo — APPROVED** [review]

**Comment by mistydemeo** [comment]
Looks like another issue:

```
16:31:31 clang -DHAVE_CONFIG_H -I. -I..  -I.. -I../autoopts  -D_THREAD_SAFE -I/usr/local/Cellar/guile/2.2.4_1/include/guile/2.2 -I/usr/local/opt/gmp/include -I/usr/local/opt/readline/include -I/usr/local/opt/bdw-gc/include -g -O2 -Wno-format-contains-nul -fno-strict-aliasing -c -o autogen-ag.o `test -f 'ag.c' || echo './'`ag.c
16:31:31 In file included from ag.c:20:
16:31:31 ./expExtract.c:48:47: error: no member named 'st_mtim' in 'struct stat'
16:31:31         if (time_is_before(outfile_time, stbf.st_mtime))
16:31:31                                          ~~~~ ^
```

**Comment by mistydemeo** [comment]
Thanks! I appreciate your patience on this.

**Comment by alebcay** [comment]
No problem! Thanks for helping out and also digging up the fixes for those issues.

**Pull request closed without merging**